test/integration/node-specific/auto_connect.test.ts

```diff
@@ -821,4 +823,80 @@ describe('When executing an operation for the first time', () => {
       });
     });
   });
+
+  describe('when CSOT is enabled', function () {
+    let client: MongoClient;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ timeoutMS: 1000 });
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    describe('when nothing is wrong', function () {
+      it('connects the client', async function () {
+        await client.connect();
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      });
+    });
+
+    describe(
+      'when the server requires auth and ping is delayed',
```

Comment:
This is slightly nit-picky, but since we don't care which parts of auto connect hang, could we simply just stub `connect()` and delay it longer than timeoutMS?

Reply:
I do agree, but since I need to keep the fail point around for the explicit connect (stubbing in that test would remove any coverage of connect accidentally respecting `timeoutMS`, you know something that is so easy to do 😅) seems easier to keep them in sync than to cover it in two different ways. Does that seem okay to you?

Reply:
I'm not sure I follow.  You're saying that test additionally is giving us coverage that connect() does not support timeoutMS?  If we want to test that connect doesn't support timeoutMS, then we should probably test that separately.

Reply:
That is what the test is testing, when the ping is delayed: client.connect() takes as long as ping is delayed for and does not throw a timeout error

It is checking that the executeOperation portion of connect which is where we handle applying a timeoutMS to an operation doesn't mistakenly pull in and apply the client's timeoutMS.

Reply:
I think we're missing a test that demonstrates:

```typescript
describe('when CSOT is enabled', function() {
  describe('when auto connect is used and connect() takes longer than timeoutMS', function() {
    it('the operation succeeds', async function() {});
  });
})
```

Or is this tested elsewhere?

Reply:
I guess this is implicitly tested in the last test block in this file.  Was that the intended CSOT + auto connect test?

Reply:
Yes it was but we can have both, it would be good to have coverage that is agnostic of the connect implementation but also include the known ping that has a risk of using timeoutMS

Reply:
Added!